torch/nn/cpp.py

```diff
@@ -25,39 +32,61 @@ def cpp_dict(self):
     # Magic methods cannot be assigned dynamically and bypass ``getattr``, so we
     # must manually override them.
 
-    def items(self):
+    def items(self) -> list[tuple[str, _T]]:  # type: ignore[override]
         return self.cpp_dict.items()
 
-    def keys(self):
+    def keys(self) -> list[str]:  # type: ignore[override]
         return self.cpp_dict.keys()
 
-    def values(self):
+    def values(self) -> list[_T]:  # type: ignore[override]
         return self.cpp_dict.values()
 
-    def __iter__(self):
+    # This should return an Iterator[str], but OrderedDict::item is not currently
+    # designed to let us iterate over only the keys.
+    def __iter__(self) -> Iterator[tuple[str, _T]]:  # type: ignore[override]
```

Comment:
Why can't we just call iter keys? or is it an API change...

Reply:
No API change; this wraps an underlying C++ custom class we've implemented, and we don't have methods available for simply looking at the keys. One could possibly be implemented, but I was more interested in documenting the existing behavior in this PR.